File: apps/frontend/src/utils/api-client/interfaces/platform.ts
/**
 * Status of the deployment-wide Rhesis platform API key.
 *
 * Only meaningful on a local/self-hosted deployment, where the prepopulated
 * Rhesis-hosted models (Rhesis, Rhesis Embedding, Rhesis
 * Polyphemus) require a Rhesis platform API key to function. Shape mirrors the
 * backend `GET/PUT/DELETE /platform/rhesis-key` response.
 */
export interface PlatformKeyStatus {
  /** Whether a platform key is currently stored. */
  configured: boolean;
  /**
   * Where the effective key comes from. Only an `organization` key can be
   * removed; an `environment` key lives in the deployment's `RHESIS_API_KEY`
   * and can only be overridden by saving an org key.
   */
  source: 'organization' | 'environment' | null;
  /** Whether the stored key validated against the Rhesis platform, or null if unchecked. */
  valid: boolean | null;
  /** Whether the stored key is authorized for Polyphemus, or null if unchecked. */
  polyphemus_authorized: boolean | null;
  /** Masked representation of the stored key (e.g. `rh_...abcd`), or null when unset. */
  masked_key: string | null;
  /** ISO timestamp of the last validation check, or null when never checked. */
  last_checked_at: string | null;
}

File: apps/backend/src/rhesis/backend/app/schemas/platform.py
"""Schemas for the local-mode Rhesis platform API key endpoints.

These back the ``/platform/rhesis-key`` router. The raw key is only ever
accepted on input (``PlatformKeyUpdate``); it is never echoed back. Status
responses expose at most a masked suffix of the stored key.
"""

from typing import Literal, Optional

from pydantic import BaseModel, Field


class PlatformKeyUpdate(BaseModel):
    """Request body for setting the org-scoped Rhesis platform API key."""

    key: str = Field(..., min_length=1, description="Rhesis platform API key (stored encrypted)")


class PlatformKeyStatus(BaseModel):
    """Status of the org-scoped Rhesis platform API key.

    Never contains the raw key -- only a masked suffix (``masked_key``) when a
    key is configured. ``valid`` and ``polyphemus_authorized`` are best-effort
    and may be ``None`` when they could not be determined.
    """

    configured: bool
    #: Where the effective key comes from. Only an ``organization`` key can be
    #: removed via DELETE; an ``environment`` key lives in the deployment's
    #: RHESIS_API_KEY and can only be overridden by storing an org key.
    source: Optional[Literal["organization", "environment"]] = None
    valid: Optional[bool] = None
    polyphemus_authorized: Optional[bool] = None
    masked_key: Optional[str] = None
    last_checked_at: Optional[str] = None
